Problem with new building system (cmake vs. gmake), CPU-features not detected

Hi,

I think I have a problem with the new building system. I'm using FreeBSD 13.3-RELEASE. A `./main`-file from November (built with gmake and Makefile) shows this

`system_info: n_threads = 7 / 8 | AVX = 1 | AVX2 = 1 | AVX512 = 0 | FMA = 1 | NEON = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | WASM_SIMD = 0 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | COREML = 0 | OPENVINO = 0 |`

whereas a current `./build/bin/main` (built with cmake) shows

`system_info: n_threads = 7 / 8 | AVX = 0 | AVX2 = 0 | AVX512 = 0 | FMA = 0 | NEON = 0 | ARM_FMA = 0 | F16C = 0 | FP16_VA = 0 | WASM_SIMD = 0 | SSE3 = 0 | SSSE3 = 0 | VSX = 0 | COREML = 0 | OPENVINO = 0 |`

So , CPU-features where not recognized properly. Can I somehow force cmake to compile with avx-, sse- and ...-flags?

Regards
abelbabel

```
$ cmake -B build-new
CMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- The C compiler identification is Clang 17.0.6
-- The CXX compiler identification is Clang 17.0.6
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/local/bin/git (found version "2.47.1")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE
-- Warning: ccache not found - consider installing it for faster compilation or disable this warning with GGML_CCACHE=OFF
-- CMAKE_SYSTEM_PROCESSOR: amd64
-- Including CPU backend
-- Found OpenMP_C: -fopenmp=libomp (found version "5.1")
-- Found OpenMP_CXX: -fopenmp=libomp (found version "5.1")
-- Found OpenMP: TRUE (found version "5.1")
-- Unknown architecture
-- Adding CPU backend variant ggml-cpu:  
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Configuring done (2.3s)
-- Generating done (0.0s)
-- Build files have been written to: /XXXXX/build-new
```

```
$ cmake --build build-new --config Release
[  3%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml.c.o
[  6%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-alloc.c.o
[  9%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-backend.cpp.o
[ 12%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-opt.cpp.o
[ 16%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-threading.cpp.o
[ 19%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-quants.c.o
[ 22%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-aarch64.c.o
[ 25%] Linking CXX shared library libggml-base.so
[ 25%] Built target ggml-base
[ 29%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.c.o
/XXXXX/whisper.cpp/ggml/src/ggml-cpu/ggml-cpu.c:247:9: warning: 'CACHE_LINE_SIZE' macro redefined [-Wmacro-redefined]
  247 | #define CACHE_LINE_SIZE 64
      |         ^
/usr/include/machine/param.h:92:9: note: previous definition is here
   92 | #define CACHE_LINE_SIZE         (1 << CACHE_LINE_SHIFT)
      |         ^
1 warning generated.
[ 32%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.cpp.o
[ 35%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu-aarch64.c.o
[ 38%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu-quants.c.o
[ 41%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/amx.cpp.o
[ 45%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/mmq.cpp.o
[ 48%] Linking CXX shared library libggml-cpu.so
[ 48%] Built target ggml-cpu
[ 51%] Building CXX object ggml/src/CMakeFiles/ggml.dir/ggml-backend-reg.cpp.o
/XXXXX/whisper.cpp/ggml/src/ggml-backend-reg.cpp:434:1: warning: non-void function does not return a value [-Wreturn-type]
  434 | }
      | ^
1 warning generated.
[ 54%] Linking CXX shared library libggml.so
[ 54%] Built target ggml
[ 58%] Building CXX object src/CMakeFiles/whisper.dir/whisper.cpp.o
[ 61%] Linking CXX shared library libwhisper.so
[ 61%] Built target whisper
[ 64%] Building CXX object examples/CMakeFiles/common.dir/common.cpp.o
/XXXXX/whisper.cpp/examples/common.cpp:266:31: warning: 'codecvt_utf8<wchar_t>' is deprecated [-Wdeprecated-declarations]
  266 |     std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
      |                               ^
/usr/include/c++/v1/codecvt:187:28: note: 'codecvt_utf8<wchar_t>' has been explicitly marked deprecated here
  187 | class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 codecvt_utf8
      |                            ^
/usr/include/c++/v1/__config:983:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
  983 | #    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
      |                                         ^
/usr/include/c++/v1/__config:956:49: note: expanded from macro '_LIBCPP_DEPRECATED'
  956 | #      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
      |                                                 ^
/XXXXX/whisper.cpp/examples/common.cpp:266:10: warning: 'wstring_convert<std::codecvt_utf8<wchar_t>>' is deprecated [-Wdeprecated-declarations]
  266 |     std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
      |          ^
/usr/include/c++/v1/locale:3598:28: note: 'wstring_convert<std::codecvt_utf8<wchar_t>>' has been explicitly marked deprecated here
 3598 | class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 wstring_convert
      |                            ^
/usr/include/c++/v1/__config:983:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
  983 | #    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
      |                                         ^
/usr/include/c++/v1/__config:956:49: note: expanded from macro '_LIBCPP_DEPRECATED'
  956 | #      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
      |                                                 ^
/XXXXX/whisper.cpp/examples/common.cpp:272:31: warning: 'codecvt_utf8<wchar_t>' is deprecated [-Wdeprecated-declarations]
  272 |     std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
      |                               ^
/usr/include/c++/v1/codecvt:187:28: note: 'codecvt_utf8<wchar_t>' has been explicitly marked deprecated here
  187 | class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 codecvt_utf8
      |                            ^
/usr/include/c++/v1/__config:983:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
  983 | #    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
      |                                         ^
/usr/include/c++/v1/__config:956:49: note: expanded from macro '_LIBCPP_DEPRECATED'
  956 | #      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
      |                                                 ^
/XXXXX/whisper.cpp/examples/common.cpp:272:10: warning: 'wstring_convert<std::codecvt_utf8<wchar_t>>' is deprecated [-Wdeprecated-declarations]
  272 |     std::wstring_convert<std::codecvt_utf8<wchar_t>> converter;
      |          ^
/usr/include/c++/v1/locale:3598:28: note: 'wstring_convert<std::codecvt_utf8<wchar_t>>' has been explicitly marked deprecated here
 3598 | class _LIBCPP_TEMPLATE_VIS _LIBCPP_DEPRECATED_IN_CXX17 wstring_convert
      |                            ^
/usr/include/c++/v1/__config:983:41: note: expanded from macro '_LIBCPP_DEPRECATED_IN_CXX17'
  983 | #    define _LIBCPP_DEPRECATED_IN_CXX17 _LIBCPP_DEPRECATED
      |                                         ^
/usr/include/c++/v1/__config:956:49: note: expanded from macro '_LIBCPP_DEPRECATED'
  956 | #      define _LIBCPP_DEPRECATED __attribute__((__deprecated__))
      |                                                 ^
4 warnings generated.
[ 67%] Building CXX object examples/CMakeFiles/common.dir/common-ggml.cpp.o
[ 70%] Building CXX object examples/CMakeFiles/common.dir/grammar-parser.cpp.o
[ 74%] Linking CXX static library libcommon.a
[ 74%] Built target common
[ 77%] Building CXX object examples/main/CMakeFiles/main.dir/main.cpp.o
[ 80%] Linking CXX executable ../../bin/main
[ 80%] Built target main
[ 83%] Building CXX object examples/server/CMakeFiles/server.dir/server.cpp.o
[ 87%] Linking CXX executable ../../bin/server
[ 87%] Built target server
[ 90%] Building CXX object examples/bench/CMakeFiles/bench.dir/bench.cpp.o
[ 93%] Linking CXX executable ../../bin/bench
[ 93%] Built target bench
[ 96%] Building CXX object examples/quantize/CMakeFiles/quantize.dir/quantize.cpp.o
[100%] Linking CXX executable ../../bin/quantize
[100%] Built target quantize
```

--- a/ggml/src/ggml-cpu/CMakeLists.txt
+++ b/ggml/src/ggml-cpu/CMakeLists.txt
@@ -175,7 +175,7 @@ function(ggml_add_cpu_backend_variant_impl tag_name)
         endif()
     elseif (CMAKE_OSX_ARCHITECTURES STREQUAL "x86_64" OR CMAKE_GENERATOR_PLATFORM_LWR MATCHES "^(x86_64|i686|amd64|x64|win32)$" OR
             (NOT CMAKE_OSX_ARCHITECTURES AND NOT CMAKE_GENERATOR_PLATFORM_LWR AND
-            CMAKE_SYSTEM_PROCESSOR MATCHES "^(x86_64|i686|AMD64)$"))
+            CMAKE_SYSTEM_PROCESSOR MATCHES "^(x86_64|i686|AMD64|amd64)$"))
         if (MSVC)
             # instruction set detection for MSVC only
             if (GGML_NATIVE)

